common/components/AddressFieldFactory/AddressFieldDropdown.tsx

```diff
@@ -0,0 +1,160 @@
+import KeyCodes from 'shared/keycodes';
+import { AppState } from 'reducers';
+import translate from 'translations';
+import { setCurrentTo, TSetCurrentTo } from 'actions/transaction';
+import { AddressLabelPair } from 'actions/addressBook';
+import { getLabels } from 'selectors/addressBook';
+import { getTransactionToRaw } from 'selectors/transactions';
+import { Address, Identicon } from 'components/ui';
+import './AddressFieldDropdown.scss';
+
+interface StateProps {
+  labels: ReturnType<typeof getLabels>;
+  currentTo: ReturnType<typeof getTransactionToRaw>;
+}
+
+interface DispatchProps {
+  setCurrentTo: TSetCurrentTo;
+}
+
+interface OwnProps {
+  onSelect(e: React.FormEvent<HTMLInputElement>): void;
+}
+
+type Props = OwnProps & StateProps & DispatchProps;
+
+interface State {
+  activeIndex: number | null;
+}
+
+class AddressFieldDropdown extends React.Component<Props> {
+  public state: State = {
+    activeIndex: null
+  };
+
+  public componentDidMount() {
+    window.addEventListener('keydown', this.handleKeyDown);
+  }
+
+  public componentWillUnmount() {
+    window.removeEventListener('keydown', this.handleKeyDown);
+  }
+
+  public render() {
+    return this.getIsVisible() ? (
+      <ul className="AddressFieldDropdown" role="listbox">
+        {this.renderDropdownItems()}
+      </ul>
+    ) : null;
+  }
+
+  private renderDropdownItems = () =>
+    this.getFilteredLabels().map((filteredLabel: AddressLabelPair, index: number) => {
+      const { activeIndex } = this.state;
+      const { address, label } = filteredLabel;
+      const isActive = activeIndex === index;
+      const className = `AddressFieldDropdown-dropdown-item ${
+        isActive ? 'AddressFieldDropdown-dropdown-item--active' : ''
+      }`;
+      const title = `${translate('SEND_TO')}${label}`;
+
+      return (
+        <li
+          key={address}
+          className={className}
+          onClick={() => this.props.setCurrentTo(address)}
+          role="option"
+          title={title}
+        >
+          <strong>{label}</strong>
+          <Identicon address={address} size="2rem" />
+          <strong>
+            <Address address={address} />
+          </strong>
+        </li>
+      );
+    });
+
+  private getFilteredLabels = () =>
+    Object.keys(this.props.labels)
+      .filter(label => label.toLowerCase().includes(this.props.currentTo.toLowerCase()))
+      .map(label => ({ address: this.props.labels[label], label }))
+      .slice(0, 5);
+
+  private getIsVisible = () =>
+    this.props.currentTo.length > 1 && this.getFilteredLabels().length > 0;
+
+  private handleKeyDown = (e: KeyboardEvent) => {
+    if (this.getIsVisible()) {
+      switch (e.keyCode) {
+        case KeyCodes.ENTER:
+          return this.handleEnterKeyDown();
+        case KeyCodes.UP_ARROW:
+          return this.handleUpArrowKeyDown();
+        case KeyCodes.DOWN_ARROW:
+          return this.handleDownArrowKeyDown();
+        default:
+          return;
+      }
```

Comment:
This is dope, make sure you also preventDefault() if it's up / down / enter.

Reply:
:+1: